src/main/java/com/github/princesslana/eriscasper/data/util/Nullable.java

```diff
@@ -0,0 +1,89 @@
+package com.github.princesslana.eriscasper.data.util;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+public class Nullable<T> {
+
+  private final Optional<T> value;
+
+  private Nullable(Optional<T> value) {
```

Comment:
Why use an Optional value for this rather than do what Optional does to store the data. Seems kind of pointless to have the item wrapped in Optional to have it as nullable.

Reply:
It would be possible, but by wrapping this in an Optional we avoid having things nullable. To me this feels consistent with the rest of the code base. Especially when we're introducing a `Nullable` class to avoid using `null`.

It also allows us to delegate many of the methods to their `Optional` equivalents, which is useful we often want Optional/Nullable to have similar behaviors.

Using a nullable field would be a possible way to implement this class also.

Reply:
Yeah I just find it awkward in some situations to have a Optional<Nullable<>> be wrapping an Optional inside of it. Makes more sense to have the item null and handle mapping features and such ourselves similar to how Optional does it. I would like to avoid `null` as much as possible but in this sense it may make sense.

Reply:
Having `Optional` inside here is only seeming confusing because you know what's inside. When using the class it doesn't matter what the implementation details are inside.

For me doing things similar to how Optional does is a reason to use Optional, because it means we don't duplicate code/logic that is already in place in Optional

Reply:
Is there any implementation of Optional left out of Nullable, it should in theory work the same exact way we just need to use it for a different purpose. I get what you're saying though, and you're right.